api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java

```diff
@@ -582,6 +594,46 @@ public int getWorkspaceCount() {
         Integer.class);
   }
 
+  @Override
+  public List<ReportingLeonardoAppUsage> getLeonardoAppUsage() {
+    if (!workbenchConfigProvider.get().reporting.exportTerraDataWarehouse) {
+      // Skip querying data if not enabled, and just return empty list instead.
+      return new ArrayList<>();
+    }
+    final QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(
+                "SELECT appId, appName, status, createdDate, destroyedDate, "
+                    + "startTime, stopTime, creator, customEnvironmentVariables "
+                    + "FROM `"
+                    + workbenchConfigProvider.get().reporting.terraWarehouseLeoAppUsageTableId
+                    + "` au "
+                    + "JOIN `"
+                    + workbenchConfigProvider.get().reporting.terraWarehouseLeoAppUsageTableId
+                    + "` a on a.id = au.appId where STARTS_WITH(appName, \"all-of-us-\")")
```

Comment:
is this `all-of-us` string defined somewhere else as well? Thinking how we can prevent breakage down the line if we decide to change naming convention for some reason down the line

Reply:
ah, good point, I will use this constant: https://github.com/all-of-us/workbench/blob/d2faf20e13a24c7627b5f148ec8fe15e62b9f4f8/api/src/main/java/org/pmiops/workbench/db/model/DbUser.java#L47

Reply:
I don't think we want to change this as app name is also GKE service name, and that is how we do egress alert